diff --git a/etc/report.d/scp.conf b/etc/report.d/scp.conf
index 4f36787..f2be585 100644
--- a/etc/report.d/scp.conf
+++ b/etc/report.d/scp.conf
@@ -1,6 +1,11 @@
 [scp]
 description="Plugin to copy report via scp"
 
+#
+# The scp plugin currently only does password authentication, all other
+# forms of authentication are disabled.
+#
+
 # plugin
 plugin=scp
 
diff --git a/python/report/plugins/scp/__init__.py b/python/report/plugins/scp/__init__.py
index 94725f8..480fe2d 100644
--- a/python/report/plugins/scp/__init__.py
+++ b/python/report/plugins/scp/__init__.py
@@ -138,9 +138,16 @@ def copyFileToRemote(exnFileName, io, optionsDict):
         raise RuntimeError("Could not fork process to run scp")
     elif childpid == 0:
         # child process - run scp
-        args = ["scp", "-oNumberOfPasswordPrompts=1",
+        args = ["scp", 
+                "-oGSSAPIAuthentication=no",
+                "-oHostbasedAuthentication=no",
+                "-oPubkeyAuthentication=no",
+                "-oChallengeResponseAuthentication=no",
+                "-oPasswordAuthentication=yes",
+                "-oNumberOfPasswordPrompts=1",
                 "-oStrictHostKeyChecking=no",
-                "-oUserKnownHostsFile=/dev/null"] + portArgs + \
+                "-oUserKnownHostsFile=/dev/null",
+                ] + portArgs + \
                [exnFileName, target]
         os.execvp("scp", args)
 
